crates/sound/src/device.rs

```diff
@@ -388,42 +388,51 @@ impl VhostUserSoundThread {
             return Ok(true);
         }
 
+        // Instead of counting descriptor chain lengths, encode the "parsing" logic in
+        // an enumeration. Then, the compiler will complain about any unhandled
+        // match {} cases if any part of the code is changed. This makes invalid
+        // states unrepresentable in the source code.
         #[derive(Copy, Clone, PartialEq, Debug)]
-        enum TxState {
+        enum IoState {
```

Comment:
This change is in preparation of Rx support, right?

Reply:
Yes but done as `sound: clean up and improve comments, definitions`